sopel/modules/instagram.py

```diff
@@ -45,43 +56,84 @@ def get_insta_json(url):
     return loads(json_astxt)
 
 
+def _get_json_data(json):
+    post_pages = json.get('entry_data', {}).get('PostPage', [])
+    if not post_pages:
+        raise ParseError('No PostPage found in %r' % json)
+    post_page = post_pages[0]
+
+    media = post_page.get('graphql', {}).get('shortcode_media', {})
+
+    if not media:
+        raise ParseError('No graphql data in %r' % post_page)
+
+    return media
+
+
 def parse_insta_json(json):
     # Parse JSON content
-    needed = json['entry_data']['PostPage'][0]['graphql']['shortcode_media']
-    iwidth = needed['dimensions']['width']
-    iheight = needed['dimensions']['height']
-    iuser = needed['owner']['username']
-    ifname = needed['owner']['full_name']
-    ilikes = needed['edge_media_preview_like']['count']
-    icomms = needed['edge_media_to_parent_comment']['count']
-    idate = needed['taken_at_timestamp']
-    pubdate = datetime.utcfromtimestamp(idate).strftime('%Y-%m-%d %H:%M:%S')
-    ivideo = needed['is_video']
-
-    # Does the post have a caption?
+    needed = _get_json_data(json)
+
+    dimensions = needed.get('dimensions', {})
+    owner = needed.get('owner', {})
+
+    # Build bot response
+    parts = []
+
+    # Title
+    if needed.get('is_video'):
+        title = "[insta] Video by "
+    else:
+        title = "[insta] Photo by "
+
+    # Author
+    iuser = owner.get('username')
+    ifname = owner.get('full_name')
+    if ifname and iuser:
+        parts.append('%s %s (@%s)' % (title, ifname, iuser))
+    elif iuser:
+        parts.append('%s @%s' % (title, iuser))
+    elif ifname:
+        parts.append('%s %s' % (title, ifname))
+    else:
+        parts.append('%s unkown user' % title)
+
+    # Media caption
     try:
         icap = needed['edge_media_to_caption']['edges'][0]['node']['text']
         # Strip newlines
         icap = icap.replace('\n', ' ')
         # Truncate caption
         icap = (icap[:256] + '…') if len(icap) > 256 else icap
-    except Exception:  # TODO: be specific
-        icap = False
-
-    # Build bot response
-    if ivideo is True:
-        botmessage = "[insta] Video by "
-    else:
-        botmessage = "[insta] Photo by "
-    if ifname is None:
-        botmessage += "@%s" % iuser
-    else:
-        botmessage += "%s (@%s)" % (ifname, iuser)
-    if icap is not False:
-        botmessage += " | " + icap
-    botmessage += " | " + str(iwidth) + "x" + str(iheight)
-    botmessage += " | Likes: {:,} | Comments: {:,}".format(ilikes, icomms)
-    botmessage += " | Uploaded: " + pubdate
-
-    # Ta-da!
-    return botmessage
+    except (KeyError, IndexError):
+        icap = None
+
+    if icap:
+        parts.append(icap)
+
+    # Media width and height
+    iwidth = dimensions.get('width') or None
+    iheight = dimensions.get('height') or None
```

Comment:
Just an interesting choice to use `or None` here instead of a default passed to `.get()`.

Reply:
An habit of mine, when I don't want to bother when `key` exists but `.get('key')` returns a false-ish value when I prefer to have `None`  instead.